packages/flutter_test/lib/src/controller.dart

```diff
@@ -879,6 +923,27 @@ class LiveWidgetController extends WidgetController {
     await binding.endOfFrame;
   }
 
+  @override
+  Future<int> pumpAndSettle([
+    Duration duration = const Duration(milliseconds: 100),
+    Duration timeout = const Duration(minutes: 10),
+  ]) {
+    assert(duration != null);
+    assert(duration > Duration.zero);
+    assert(timeout != null);
+    assert(timeout > Duration.zero);
+    int count = 0;
+    return TestAsyncUtils.guard<void>(() async {
+      final DateTime endTime = clock.fromNowBy(timeout);
+      do {
+        if (clock.now().isAfter(endTime))
+          throw FlutterError('pumpAndSettle timed out');
```

Comment:
We should remove this timeout feature, per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#never-check-if-a-port-is-available-before-using-it-never-add-timeouts-and-other-race-conditions

Maybe just make the default null and assert that it is null in this version, and mark it deprecated in the superclass and the other implementation? Or maybe remove it entirely? Could be another PR to do that... I'd rather we didn't add more code that supported it though.

Reply:
Removed timeout.